MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm

```diff
@@ -48,12 +48,9 @@
 		MVKDescriptorSet* descSet = descriptorSets[dsIdx];
 		uint32_t dslIdx = firstSet + dsIdx;
 		MVKDescriptorSetLayout* dsl = _descriptorSetLayouts[dslIdx];
-		dynamicOffsetIndex += dsl->bindDescriptorSet(cmdEncoder, descSet,
-													 _dslMTLResourceIndexOffsets[dslIdx],
-													 dynamicOffsets, dynamicOffsetIndex);
-		if (!cmdEncoder && dsl->getConfigurationResult() != VK_SUCCESS) {
-			setConfigurationResult(dsl->getConfigurationResult());
-		}
+		dsl->bindDescriptorSet(cmdEncoder, descSet, _dslMTLResourceIndexOffsets[dslIdx],
+							   dynamicOffsets, dynamicOffsetIndex);
+		if (!cmdEncoder) { setConfigurationResult(dsl->getConfigurationResult()); }
```

Comment:
Wait, if one fails, but a later one doesn't, won't that cause the whole thing to report success? There's a reason I put that `dsl->getConfigurationResult() != VK_SUCCESS` check in.

Reply:
Yeah...if figured the loop was why you added that test.

`setConfigurationResult()` is a latch function. It will only move from `VK_SUCCESS` to something else. The design idea is intentionally to catch and hold onto the first error that occurs (loop or not).

Reply:
Oh, OK. I did not know that.